src/Subscriptions/Storage/RedisStorageManager.php

```diff
@@ -0,0 +1,169 @@
+    public function subscribersByTopic(string $topic): Collection
+    {
+        $subscriberIds = $this->connection->command('smembers', [$this->topicKey($topic)]);
+        $subscriberIds = array_map([$this, 'channelKey'], $subscriberIds);
+        $subscribers = $this->connection->command('mget', [$subscriberIds]);
+
+        return (new Collection($subscribers))
+            ->map(function ($subscriber) {
+                return $this->unserialize($subscriber);
+            })
+            ->filter();
+    }
+
+    public function storeSubscriber(Subscriber $subscriber, string $topic): void
+    {
+        $subscriber->topic = $topic;
+
+        $topicKey = $this->topicKey($topic);
+        $this->connection->command('sadd', [
+            $topicKey,
+            $subscriber->channel,
+        ]);
+        if (isset($this->ttl)) {
+            $this->connection->command('expire', [$topicKey, $this->ttl]);
+        }
+
+        $this->connection->command('set', array_merge([
+            $this->channelKey($subscriber->channel),
+            $this->serialize($subscriber),
+        ], $this->ttl ? ['EX', $this->ttl] : []));
+    }
+
+    /**
+     * @return Subscriber|null
+     */
+    public function deleteSubscriber(string $channel)
+    {
+        $key = $this->channelKey($channel);
+        $subscriber = $this->getSubscriber($key);
+
+        if ($subscriber) {
+            $this->connection->command('del', [$key]);
+            $this->connection->command('srem', [
+                $this->topicKey($subscriber->topic),
+                $channel,
+            ]);
+        }
+
+        return $subscriber;
+    }
+
+    /**
+     * @param string $channelKey
+     */
+    protected function getSubscriber($channelKey): ?Subscriber
+    {
+        $subscriber = $this->unserialize(
+            $this->connection->command('get', [$channelKey])
+        );
+
+        if ($subscriber instanceof Subscriber) {
+            return $subscriber;
+        }
+
+        return null;
+    }
+
+    protected function channelKey(string $channel): string
+    {
+        return self::SUBSCRIBER_KEY.'.'.$channel;
+    }
+
+    protected function topicKey(string $topic): string
+    {
+        return self::TOPIC_KEY.'.'.$topic;
+    }
+
+    /**
+     * @param mixed $value Value to serialize.
+     * @return mixed Storable value.
+     * @see \Illuminate\Cache\RedisStore::serialize
+     */
+    protected function serialize($value)
+    {
+        $isProperNumber = is_numeric($value)
+            && ! in_array($value, [INF, -INF])
+            && ! is_nan(floatval($value));
+
+        return $isProperNumber
+            ? $value
+            : serialize($value);
+    }
+
+    /**
+     * @param mixed $value Value to unserialize.
+     * @return mixed Unserialized value.
+     */
+    protected function unserialize($value)
+    {
+        if (false === $value) {
+            return null;
+        }
+
+        return is_numeric($value) ? $value : unserialize($value);
```

Comment:
```suggestion
        return is_numeric($value)
            ? $value
            : unserialize($value);
```

We always use multiline ternarys.